### PR TITLE
Bug 1880075 - Use single name field for addresses

### PR DESCRIPTION
**Pull request by issammani**
This PR (is):
- Changes address types to use a single name field instead of 3 `*-name` fields.
- Fixes failing tests
- Changes address edit menu to have a single input field instead of 3.
- Blocked by https://github.com/mozilla/application-services/pull/6118
- Changes UI to have one single name field similar  to Desktop
Screenshot 2024-02-13 at 15 52 59

## ⚠️ NOTES
I built the a-s change locally:
- both fenix and android-components are building 
- all tests are passing with the exception of these tests that were also failing before the change ( Let me know if we have some CI to test this with the a-s change):

  -  DefaultPrivateBrowsingControllerTest. WHEN private browsing learn more link is clicked THEN open support page in browser
  - GeckoEngineTest. test parsing engine version
  - SearchActionProviderTest. provider returns suggestion matching input
  - AsyncFilterListenerTest. filter delegate checks for cancellations before it runs, passes results to autocomplete view
  - CustomTabsToolbarFeatureTest. close button invokes callback and removes session

### Pull Request checklist

- [ ] ~**Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)~
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] ~**Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one~
- [ ] ~**Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features~

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1880075

### CODE REVIEW

**Comment by issammani** [comment]
cc @dimidl

**Comment by jeffreygee** [comment]
@issammani - no issues with UX here from me, looks consistent with Desktop direction.

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuAutofillRobot.kt
Comment (MatthewTighe) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuAutofillRobot.kt:149]: Just looking over this, I'm wondering if we'll hit issues with this or other UI tests due to changes in the visibility of expected items. I think if we do hit any issues, we can just `@Ignore` the tests to land the patch and then do a quick follow-up to address the broken tests.

Once this is approved, please double-check with either me or another Fenix engineer if you end up needing to `@Ignore` any tests so that we can help verify that they are safe to temporarily ignore. If it's time-sensitive, @t-p-white and @rahulsainani would be good options that look closer to your timezone as its shown on Slack.

File: fenix/app/src/main/res/values/strings.xml
Comment (MatthewTighe) [inline fenix/app/src/main/res/values/strings.xml:1837]: We have a [specific process](https://github.com/mozilla-mobile/fenix/blob/main/docs/Working-with-Strings.md) for removing strings that we'll need to follow instead of just deleting these.

**Review by MatthewTighe — CHANGES_REQUESTED** [review]
Great work, thanks for handling this update! Just a small change needed to how the strings in `strings.xml` are being removed.

File: fenix/app/src/main/res/values/strings.xml
Comment (issammani) [reply fenix/app/src/main/res/values/strings.xml]: TIL. Thanks. This is fixed in [7dd261c](https://github.com/mozilla-mobile/firefox-android/pull/5586/commits/7dd261cf41a513b1a49419016ba0a0b90d67629f)

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuAutofillRobot.kt
Comment (issammani) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuAutofillRobot.kt]: Makes sense. Once this PR is approved I can land https://github.com/mozilla/application-services/pull/6118. Then I will ask for your help ( or team ) regarding this. Thanks for the heads-up.

**Review by MatthewTighe — APPROVED** [review]
LGTM! Let's wait until Monday to start the process of landing the A-S patch and this one so we don't get stuck on them over the weekend

File: android-components/components/concept/storage/src/main/java/mozilla/components/concept/storage/CreditCardsAddressesStorage.kt
Comment (t-p-white) [inline android-components/components/concept/storage/src/main/java/mozilla/components/concept/storage/CreditCardsAddressesStorage.kt:354]: Could we update this from `name` to `fullName` or `completeName`? At a minimum I think we should update the description to something such as "A person's full name, typically made up of a first name, any middle names and surname, e.g. John Joe Doe"

**Review by t-p-white — COMMENTED** [review]
Could we update all `name` descriptions to something such as "A person's full name, typically made up of a first name, any middle names and surname, e.g. John Joe Doe". Currently, it's a little vague & open to interpretation imo

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuAutofillRobot.kt
Comment (issammani) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuAutofillRobot.kt]: @MatthewTighe @t-p-white a-s PR is landed and I updated the nightly version. Waiting for CI to finish. Might need your help now if any tests fail or if I need to add `@Ignore` anywhere 😄

File: fenix/app/src/main/res/values/strings.xml
Comment (gabrielluong) [inline fenix/app/src/main/res/values/strings.xml:1851]: @issammani These `moz:removedIn` actually need to be bumped to 125 since we are currently in Nightly 125 https://whattrainisitnow.com/. Would you be able to make a followup change? Thanks

cc @MatthewTighe @t-p-white

Comment (issammani) [reply fenix/app/src/main/res/values/strings.xml:1851]: Sorry my bad. In my mind it's still 124 for some reason. Anyways, this PR https://github.com/mozilla-mobile/firefox-android/pull/5776 should fix it 😄

**Pull request merged**
